client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -80,9 +89,13 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 		};
 
 		submitForm = () => {
-			const { fields, site } = this.props;
+			const { fields, settingsFields, site, isJetpack } = this.props;
 			this.props.removeNotice( 'site-settings-save' );
-			this.props.saveSiteSettings( site.ID, fields );
+
+			this.props.saveSiteSettings( site.ID, pick( fields, settingsFields.site ) );
```

Comment:
I was wondering how you would handle the distinction between the settings. This is wise. Though, it will work only if we don't have a common key on both settings.